command/pr_review.go

```diff
@@ -0,0 +1,137 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cli/cli/api"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	prCmd.AddCommand(prReviewCmd)
+
+	prReviewCmd.Flags().StringP("approve", "a", "", "Approve pull request")
+	prReviewCmd.Flags().StringP("request-changes", "r", "", "Request changes on a pull request")
+	prReviewCmd.Flags().StringP("comment", "c", "", "Comment on a pull request")
+
+	// this is required; without it pflag complains that you must pass a string to string flags.
+	prReviewCmd.Flags().Lookup("approve").NoOptDefVal = " "
+	prReviewCmd.Flags().Lookup("request-changes").NoOptDefVal = " "
+	prReviewCmd.Flags().Lookup("comment").NoOptDefVal = " "
+}
+
+var prReviewCmd = &cobra.Command{
+	Use:   "review",
+	Short: "TODO",
+	Args:  cobra.MaximumNArgs(1),
+	Long:  "TODO",
+	RunE:  prReview,
+}
+
+func processReviewOpt(cmd *cobra.Command) (*api.PullRequestReviewInput, error) {
+	found := 0
+	flag := ""
+	var state api.PullRequestReviewState
+
+	if cmd.Flags().Changed("approve") {
+		found++
+		flag = "approve"
+		state = api.ReviewApprove
+	}
+	if cmd.Flags().Changed("request-changes") {
+		found++
+		flag = "request-changes"
+		state = api.ReviewRequestChanges
+	}
+	if cmd.Flags().Changed("comment") {
+		found++
+		flag = "comment"
+		state = api.ReviewComment
+	}
+
+	if found != 1 {
+		return nil, errors.New("need exactly one of --approve, --request-changes, or --comment")
+	}
+
+	val, err := cmd.Flags().GetString(flag)
+	if err != nil {
+		return nil, err
+	}
+
+	body := ""
+	if val != " " {
+		body = val
+	}
+
+	if flag == "comment" && (body == " " || len(body) == 0) {
```

Comment:
We might consider extending this behavior to `--request-changes` too, since it's not possible to request changes without leaving a comment either.

Since `--approve` is the only option where comment body is optional, we could consider marking other flags as always requiring a value.

Reply:
Ah, good catch. I'll fix that validation during retooling the flags.